### ISSUE TITLE
Problem with Saving

### ISSUE THREAD

**Issue by dickboner93** [stolksdorf/naturalcrit#100]
Hey there!

First off, thank you for making an amazing tool that is easy to use for the masses.

To get to the point, I have been having issues saving my files over time. Long story short, I am using the homebrew tool to make a player's handout for an upcoming home game and there are a lot of pages and chapters. Every now and then, when I try to copy and paste from a different source (from other homebrew files I made with natural crit), the Save option is in a constant "saving..." state, and never eventually saves. Even typing in everything from another source is causing this problem to happen.

Am I the only one? Or is this happening to others as well?

Here is my brew if needed:
http://www.naturalcrit.com/homebrew/edit/Hyg-UYjFf

* Browser: Google Chrome Version 50.0.2661.102
* Operating System: Windows 8.1

**Comment by stolksdorf** [comment]
I know exactly what's happening! Very rarely my error logs will show a bunch 413 errors (entity too large). I disregarded it because the amount of text you need to send in a request is _massive_ to trigger that, so I thought it was a bug or someone trying to upload something to my server or whatever.

I can roll out a fix pretty quick. I'm just going to increase the entity maximum limit. 

Can you send me the share id of some of your brews that are affected by this?

**Comment by dickboner93** [comment]
Yea, here is essentially what I'm working with:
(Main Chapter to transfer all files to)
http://www.naturalcrit.com/homebrew/edit/Hyg-UYjFf
(Files needed to be transfered, if you need access to those as well)
http://www.naturalcrit.com/homebrew/edit/H1xhUQnMz

http://www.naturalcrit.com/homebrew/edit/HylH5ZcXz

http://www.naturalcrit.com/homebrew/edit/BkzCt1r4G

http://www.naturalcrit.com/homebrew/edit/BJeZNdY4f

http://www.naturalcrit.com/homebrew/edit/HJxo_IcDf

http://www.naturalcrit.com/homebrew/edit/BJerXvEtM

http://www.naturalcrit.com/homebrew/edit/HJlVALOdG

http://www.naturalcrit.com/homebrew/edit/H1eiH8tdG

http://www.naturalcrit.com/homebrew/edit/rklsp_LHG

http://www.naturalcrit.com/homebrew/edit/r1ejPKtOM

http://www.naturalcrit.com/homebrew/edit/B1fyd29uM

http://www.naturalcrit.com/homebrew/edit/BygyUkzFz

http://www.naturalcrit.com/homebrew/edit/SkgiDemtf

http://www.naturalcrit.com/homebrew/edit/ryeEdKYYz

**Comment by stolksdorf** [comment]
Thanks! I'm going to try and sneak in a bit of a code session before work so I can roll out a fix for you.

**Comment by dickboner93** [comment]
Awesome thank you so much. I will by you a lot of beer for your hard work!

**Comment by stolksdorf** [comment]
Ah so you are copying each of those into the main one? Okay this is starting to make sense of the sheer size.  You are intense.I never expected someone to use my tool like this!

**Comment by dickboner93** [comment]
Sorry lol. In essence I loved your tool so much I wanted to make a handout to eventual "source book" for the homebrew I've been working on for a while.

Eventually after I had all the chapters finished using your tool, I would download them as pdfs (with printer friendly versions) and combine them to one to save the tool some more time for rendering and send to my players.

**Comment by stolksdorf** [comment]
Don't be sorry, this is awesome. You're the exact person I wrote this tool for.

I was able to whip up the solution, just stressed tested it at 1200 pages and it saved like a charm. I'm going to quickly polish the code and add on some error handling for saving.

**Comment by dickboner93** [comment]
You're the man! When I have time later today, prepare to have a new patron on Patreon.

**Comment by stolksdorf** [comment]
Tested on staging and just deploying to prod now. I added a little error box that pops up with a prefilled-out link to submit a github issue with the contents of the error message from my server. Should make reporting these issues better in the future for both the user and me.

The payload limit is now 25mb, which should be plenty. If you run into this error again give me a heads up and I can bump it further :)

Happy brewing!
